File: js/hanoi.mjs
function hanoi(count, from, to, temp) {
    if (count == 0) {
        return;
    }
    
    hanoi(count - 1, from, temp, to);
    console.log(`원반 ${count}를 ${from}에서 ${to}로 이동`)
    hanoi(count - 1, temp, to, from);
}

// 원반 개수, 각 기둥
hanoi(3, "A", "B", "C")

File: python/Hanoi.py
def hanoi(count, source, target, temp):
    if count == 0:
        return
    hanoi(count - 1, source, temp, target)
    print(f"{count} : {source} -> {target}")
    hanoi(count - 1, temp, target, source)

hanoi(3, "A", "C", "B")